### PR TITLE
Automatically configure a VM endpoint

### LINKED ISSUE #184: Configure mesh expansion with the repository and onboarding plane endpoint
Today we generate an empty mesh expansion configuration in the CPs.
We could easily add some config bits to allow configuring which clusters should have mesh expansion enabled, and we could also generate a proper FQDN for the vmgateway (something like `vm<cluster_id>-<tsb_fqdn>`), configure it in the provider DNS, and generate the corresponding certificate, so that the cluster is ready to be used to onboard VMs.

I have a WIP branch for this and plan to submit a PR soon.

### PR DESCRIPTION
**Pull request by nacx**
Fixes https://github.com/tetrateio/tetrate-service-bridge-sandbox/issues/184

Now that we can install external-dns we can automatically configure the VM endpoint.

### CODE REVIEW

File: modules/tsb/cp/main.tf [hunk 1]
```diff
@@ -89,6 +89,7 @@ resource "helm_release" "controlplane" {
     ratelimit_enabled               = var.ratelimit_enabled
     ratelimit_namespace             = var.ratelimit_namespace
     identity_propagation_enabled    = var.identity_propagation_enabled
+    vm_endpoint                     = "vms.${var.cluster_name}.${regep("[^\\.]*\\.(.*)", var.tsb_fqdn)}"
```
Comment (nacx) [inline modules/tsb/cp/main.tf]: the regexp is basically removing the first block to capture the DNS suffix

File: modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl [hunk 2]
```diff
@@ -78,7 +87,13 @@ spec:
...
           uri: redis-master.${ratelimit_namespace}.svc.cluster.local:6379
     %{ endif }
-  meshExpansion: {}
+  meshExpansion:
+    onboarding:
+      localRepository: {}
+      endpoint:
+        hosts:
+          - ${vm_endpoint}
+        secretName: vm-onboarding
```
Comment (nacx) [inline modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl:96]: Opened as draft to see if it is OK to configure this via ext-dns. If the approach is OK I'll add the bots to generate the certificate and populate this secret, which is still not in this PR.

File: modules/tsb/cp/main.tf (hunk 1 above)
Comment (smarunich) [reply modules/tsb/cp/main.tf]: to keep it generic... should we have a vm_endpoint as a variable for tsb_cp module ... $var.tsb_fqdn may be or can be different from external-dns domain... so basically under module itself... we should define vm_endpoint as

vm_endpoint = $var.cluster_name + lookup/element to select external_dns_$cloud_dns_zone

File: modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl (hunk 2 above)
Comment (smarunich) [reply modules/tsb/cp/manifests/tsb/controlplane-values.yaml.tmpl:96]: oh yeah, so heavily looking forward to the trusted certificate :)

File: tsb/cp/variables.tf [hunk 3]
```diff
@@ -139,4 +139,8 @@ variable "ratelimit_enabled" {
 
 variable "identity_propagation_enabled" {
   default = false
-}
\ No newline at end of file
+}
+
+variable "vm_endpoint" {
+  default = "vms.${var.cluster_name}.${regep("[^\\.]*\\.(.*)", var.tsb_fqdn)}"
```
Comment (smarunich) [inline tsb/cp/variables.tf]: regep > `regexp`

**Pull request closed without merging**